presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/PlanMatchPattern.java

```diff
@@ -784,6 +785,12 @@ public PlanMatchPattern withOutputRowCount(double expectedOutputRowCount)
         return this;
     }
 
+    public PlanMatchPattern withSourceInfoAndConfidenceLevel(SourceInfo sourceInfo)
```

Comment:
I think as written, I would just call this `withSourceInfo()`. however, HistoryBasedSourceInfos are a bit hard to generate for tetsts, so it might be more useful to have two matchers that check what we are most interested in `withSourceInfoType(String expectedSourceInfoName)` that compares that the sourceInfo.getName() matches the expected and another `withConfidenceLevel(ConfidenceLevel expectedConfidenceLevel)` that checks that the confidence level matches the expected.